concordium-consensus/src/Concordium/TransactionVerification.hs

```diff
@@ -0,0 +1,253 @@
+{-# LANGUAGE TypeFamilies #-}
+module Concordium.TransactionVerification
+  where
+
+import Control.Monad.Trans
+import Control.Monad.Trans.Reader
+import qualified Data.Map.Strict as OrdMap
+import qualified Data.Serialize as S
+
+import qualified Concordium.Types.Transactions as Tx
+import qualified Concordium.ID.AnonymityRevoker as AR
+import qualified Concordium.ID.IdentityProvider as IP
+import qualified Concordium.GlobalState.Types as GSTypes
+import qualified Concordium.GlobalState.BlockState as BS
+import qualified Concordium.Types.Parameters as Params
+import qualified Concordium.Types.Updates as Updates
+import qualified Concordium.Types as Types
+import qualified Concordium.ID.Account as A
+import qualified Concordium.ID.Types as ID
+import qualified Concordium.Crypto.SHA256 as Sha256
+import qualified Concordium.Scheduler.Types as ST
+import Concordium.Types.HashableTo (getHash)
+import qualified Concordium.Cost as Cost
+
+import Data.Maybe (isJust)
+import Control.Monad.Except
+
+-- |The 'VerificationResult' type serves as an intermediate `result type` between the 'TxResult' and 'UpdateResult' types.
+-- VerificationResult's contains possible verification errors that may have occurred when verifying a 'AccountCreation' type.
+data VerificationResult
+  = CredentialDeploymentSuccess
+  -- ^The 'CredentialDeployment' passed verification.
+  | CredentialDeploymentDuplicateAccountRegistrationID !ID.CredentialRegistrationID
```

Comment:
Whether or not this should be considered maybeValid depends on which block we validate it against. If it is last finalized it is OK to not consider this maybe valid, but if it is focus or best then it is not.